diff --git a/.gitignore b/.gitignore
new file mode 100644
index 0000000..9dd52c6
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1 @@
+asciiportal
diff --git a/Makefile.default b/Makefile.default
index f9b0be2..31d4843 100644
--- a/Makefile.default
+++ b/Makefile.default
@@ -125,14 +125,14 @@ distclean:
 # for an example
 
 install: saveclean
-	mkdir -p $(DESTDIR)/usr/share/asciiportal
-	mkdir -p $(DESTDIR)/usr/share/doc/asciiportal
-	mkdir -p $(DESTDIR)/usr/bin
-	cp -r media/ maps/ $(DESTDIR)/usr/share/asciiportal/
-	cp README CHANGELOG $(DESTDIR)/usr/share/doc/asciiportal/
-	cp $(EXE_NAME) $(DESTDIR)/usr/bin/
-	chmod 0644 $(DESTDIR)/usr/share/doc/asciiportal/*
-	chmod 0755 $(DESTDIR)/usr/bin/$(EXE_NAME)
+	mkdir -p $(DESTDIR)/share/asciiportal
+	mkdir -p $(DESTDIR)/share/doc/asciiportal
+	mkdir -p $(DESTDIR)/bin
+	cp -r media/ maps/ $(DESTDIR)/share/asciiportal/
+	cp README CHANGELOG $(DESTDIR)/share/doc/asciiportal/
+	cp $(EXE_NAME) $(DESTDIR)/bin/
+	chmod 0644 $(DESTDIR)/share/doc/asciiportal/*
+	chmod 0755 $(DESTDIR)/bin/$(EXE_NAME)
 
 $(TARBALL_SRC):
 	mkdir $(DISTNAME)-src
@@ -168,4 +168,4 @@ yaml-cpp: yaml-cpp-$(YAML-CPP_VER).tar.gz
 	tar xvfz $<
 
 yaml-cpp-$(YAML-CPP_VER).tar.gz:
-	wget http://yaml-cpp.googlecode.com/files/yaml-cpp-$(YAML-CPP_VER).tar.gz
\ No newline at end of file
+	wget http://yaml-cpp.googlecode.com/files/yaml-cpp-$(YAML-CPP_VER).tar.gz
diff --git a/Makefile.linux b/Makefile.linux
index b9628ca..50b00c3 100644
--- a/Makefile.linux
+++ b/Makefile.linux
@@ -18,11 +18,11 @@ DISTNAME := $(DISTNAME)-$(PORT)64
 
 # Custom C flags: you may define ('-D') or unset ('-U') macros to get your specific
 # code to work.
-CXXFLAGS := $(CXXFLAGS) -I/usr/include/xcurses `sdl-config --cflags` -U__NOSDL__ -U__NOSOUND__ -U__DINGOO__ -U__GP2X__
+CXXFLAGS := $(CXXFLAGS) -I$(DESTDIR)/include -I/usr/include/xcurses `sdl-config --cflags` `pkg-config SDL_mixer --cflags` -DDESTDIR=$(DESTDIR) -U__NOSDL__ -U__NOSOUND__ -U__DINGOO__ -U__GP2X__ -O0 -w
 
 # Custom link flags: either dynamic (default) or static linking (see the last form,
 # 'man ld' for details).
-LINKFLAGS := $(LINKFLAGS) `sdl-config --libs` -L/usr/lib -l SDL_mixer -l pdcurses
+LINKFLAGS := $(LINKFLAGS) `sdl-config --libs` -L$(DESTDIR)/lib -L/usr/lib -l SDL_mixer -l pdcurses
 
 # Additional files to be included in release tarball.
 #FILES_INCLUDE_DIST := $(FILES_INCLUDE_DIST) custom_readme.txt example.dll foo/bar.xml
@@ -59,6 +59,9 @@ wrong-way:
 	@echo "[*] Instead, call the main Makefile with the proper target!"
 	@echo "[*] In most cases, a simple 'make' should work though."
 
+print-cxxflags:
+	@echo $(CXXFLAGS)
+
 # Targets called by the main Makefile:
 $(PORT):
 	$(MAKE) -f Makefile.$(PORT) clean execlean
diff --git a/src/ap_draw.cpp b/src/ap_draw.cpp
index 92a15f0..57c2a5e 100644
--- a/src/ap_draw.cpp
+++ b/src/ap_draw.cpp
@@ -45,7 +45,7 @@ using namespace std;
 #include "pdcsdl.h"
 #endif
 
-#include <curses.h>
+#include <pdcurses.h>
 
 #ifndef __NOSOUND__
 #include "ap_sound.h"
diff --git a/src/ap_draw.h b/src/ap_draw.h
index 6143566..02a86ec 100644
--- a/src/ap_draw.h
+++ b/src/ap_draw.h
@@ -26,7 +26,7 @@
 #define AP_DRAW_H_INCLUDED
 
 #include <vector>
-#include <curses.h>
+#include <pdcurses.h>
 
 #include "asciiportal.h"
 
diff --git a/src/ap_filemgr.cpp b/src/ap_filemgr.cpp
index 6e74cf5..cc26819 100644
--- a/src/ap_filemgr.cpp
+++ b/src/ap_filemgr.cpp
@@ -41,6 +41,11 @@ using namespace std;
 #include "asciiportal.h"
 #include "ap_filemgr.h"
 
+// If compiled with -DDESTDIR=/foo
+// then STRINGIFY(DESTDIR) will expand to "/foo"
+#define STRINGIFY2(X) #X
+#define STRINGIFY(X) STRINGIFY2(X)
+
 string FileManager::inscreen = "inscreen.txt";
 string FileManager::credits = "credits.txt";
 string FileManager::infos = "infos.yaml";
@@ -129,8 +134,8 @@ FileManager::FileManager() {
   s = "/";
 
   // Search for a decent basepath for media and maps (on unix)
-  if (file_exists("/usr/share/asciiportal"))
-    basepath = "/usr/share/asciiportal";
+  if (file_exists(STRINGIFY(DESTDIR) "/share/asciiportal"))
+    basepath = STRINGIFY(DESTDIR) "/share/asciiportal";
 #endif
 
   // Look for a user-specific directory to store save data and custom maps
diff --git a/src/ap_input.cpp b/src/ap_input.cpp
index 177ae45..448706f 100644
--- a/src/ap_input.cpp
+++ b/src/ap_input.cpp
@@ -22,7 +22,7 @@
 // The source links with SDL, PDCurses compiled for SDL rendering (PDCSDL)
 // and SDL mixer.
 
-#include <curses.h>
+#include <pdcurses.h>
 #include <cstdlib>
 #include <iostream>
 #include <fstream>
diff --git a/src/ap_pager.h b/src/ap_pager.h
index 5bc531d..da8a683 100644
--- a/src/ap_pager.h
+++ b/src/ap_pager.h
@@ -28,7 +28,7 @@
 #include <utility>
 #include <list>
 #include <string>
-#include <curses.h>
+#include <pdcurses.h>
 
 #include "asciiportal.h"
 
diff --git a/src/ap_play.cpp b/src/ap_play.cpp
index 4ebbf56..cbba293 100644
--- a/src/ap_play.cpp
+++ b/src/ap_play.cpp
@@ -22,7 +22,7 @@
 // The source links with SDL, PDCurses compiled for SDL rendering (PDCSDL)
 // and SDL mixer.
 
-#include <curses.h>
+#include <pdcurses.h>
 #include <cstdlib>
 #include <iostream>
 #include <fstream>
diff --git a/src/ap_sound.cpp b/src/ap_sound.cpp
index 814c7ba..ad6548e 100644
--- a/src/ap_sound.cpp
+++ b/src/ap_sound.cpp
@@ -24,7 +24,7 @@
 
 #include <string>
 #include <iostream>
-#include <curses.h>
+#include <pdcurses.h>
 
 #include "ap_sound.h"
 
diff --git a/src/menu.cpp b/src/menu.cpp
index 3885b15..5fffc29 100644
--- a/src/menu.cpp
+++ b/src/menu.cpp
@@ -22,7 +22,7 @@
 // The source links with SDL, PDCurses compiled for SDL rendering (PDCSDL)
 // and SDL mixer.
 
-#include <curses.h>
+#include <pdcurses.h>
 #include <cstdlib>
 #include <string>
 #include <vector>
