diff -Naur hdump-2.3/Makefile hdump-2.3-dirt/Makefile
--- hdump-2.3/Makefile	2014-04-08 18:33:08.000000000 -0400
+++ hdump-2.3-dirt/Makefile	2022-11-18 12:35:47.541803259 -0500
@@ -1,11 +1,14 @@
+PREFIX=/usr/local
+
 all:
 	gcc -O2 -W -Wall -Wextra -ansi -pedantic -o hdump hdump.c
 
 install:
-	install -m 0755 hdump /usr/local/bin/hdump
+	mkdir -p $(PREFIX)/bin
+	install -m 0755 hdump $(PREFIX)/bin/hdump
 
 uninstall:
-	rm -f /usr/local/bin/hdump
+	rm -f $(PREFIX)/bin/hdump
 
 clean:
 	rm -f hdump
